mackerel-plugin-gcp-compute-engine/lib/gcp-compute-engine.go

```diff
@@ -0,0 +1,202 @@
+		},
+	},
+}
+
+// GraphDefinition is return graphdef
+func (p ComputeEnginePlugin) GraphDefinition() map[string]mp.Graphs {
+	return graphdef
+}
+
+func getLatestValue(listCall *monitoring.ProjectsTimeSeriesListCall, filter string, startTime string, endTime string, opts *Option) (float64, error) {
+	res, err := listCall.Filter(filter).IntervalEndTime(endTime).IntervalStartTime(startTime).Do(*opts)
+
+	if err != nil || res == nil {
+		return 0, err
+	}
+
+	valuePtr := res.TimeSeries[0].Points[0].Value
+
+	var value float64
+	if valuePtr.Int64Value != nil {
+		value = float64(*valuePtr.Int64Value)
+	} else if valuePtr.DoubleValue != nil {
+		value = *valuePtr.DoubleValue
+	}
+
+	return value, nil
+}
+
+func mkFilter(domain string, metricName string, instance string) string {
+	filter := `metric.type = "` + domain + metricName + `"`
+	switch domain {
+	case computeDomain:
+		filter += " AND metric.label.instance_name = " + instance
+	}
+
+	return filter
+}
+
+// FetchMetrics fetches metrics from Google Monitoring API
+func (p ComputeEnginePlugin) FetchMetrics() (map[string]float64, error) {
+	now := time.Now()
+	formattedEnd := now.Format(zuluFormat)
+	m, _ := time.ParseDuration(duration)
+	formattedStart := now.Add(-m).Format(zuluFormat)
+	listCall := p.MonitoringService.Projects.TimeSeries.List(p.Project)
+
+	stat := map[string]float64{}
+	for _, metricName := range []string{
+		"/firewall/dropped_bytes_count",
+		"/firewall/dropped_packets_count",
+		"/instance/cpu/utilization",
+		"/instance/disk/read_bytes_count",
+		"/instance/disk/read_ops_count",
+		"/instance/disk/write_bytes_count",
+		"/instance/disk/write_ops_count",
+		"/instance/network/received_bytes_count",
+		"/instance/network/received_packets_count",
+		"/instance/network/sent_bytes_count",
+		"/instance/network/sent_packets_count",
+	} {
+		value, err := getLatestValue(listCall, mkFilter(computeDomain, metricName, p.InstanceName), formattedStart, formattedEnd, p.Option)
+		if err != nil {
+			continue
+		}
+		splited := strings.Split(metricName, "/")
+		stat[splited[len(splited) - 1]] = value
+	}
+
+	return stat, nil
+}
+
+// Do the plugin
+func Do() {
+	optProject := flag.String("project", "", "Project No")
+	optInstanceName := flag.String("instance-name", "", "Instance Name")
+	optAPIKey := flag.String("api-key", "", "API key")
+	flag.Parse()
+
+	if *optProject == "" || *optInstanceName == "" || *optAPIKey == "" {
+		fmt.Println("Errors:", errors.New("Not enough arguments"))
+		return
+	}
+
+	ctx := context.Background()
+
+	client, err := google.DefaultClient(ctx, monitoring.CloudPlatformScope)
+	if err != nil {
+		return
+	}
+
+	service, err := monitoring.New(client)
+	if err != nil {
+		return
+	}
+
+	var computeEngine = ComputeEnginePlugin{
+		MonitoringService: service,
+		Project:           "projects/" + *optProject,
+		InstanceName:      *optInstanceName,
+		Option:            &Option{Key: *optAPIKey},
+	}
+
+	helper := mp.NewMackerelPlugin(computeEngine)
```

Comment:
Though not necessary, we recommend to implement `-tempfile` option to allow users to change tempfile path: https://github.com/mackerelio/go-mackerel-plugin/blob/master/mackerel-plugin.go#L47